server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -181,6 +181,22 @@ private void synchronizeDeprecatedKeys(DbSession dbSession, List<RulesDefinition
       .flatMap(rule -> SingleDeprecatedRuleKey.from(rule).stream())
       .collect(MoreCollectors.toSet());
 
+    Set<RuleKey> currentKeys = repositories.stream()
```

Comment:
I was expecting the implementation of this ticket to be based on a `Multiset` where any entry which count was > 1 would cause failure

however, this work as well